packages/aws-cdk-lib/aws-ec2/lib/vpc.ts

```diff
@@ -2637,6 +2612,15 @@ class ImportedSubnet extends Resource implements ISubnet, IPublicSubnet, IPrivat
     return this._ipv4CidrBlock;
   }
 
+  public get routeTable(): IRouteTable {
+    if (!this._routeTableId) {
+      throw new Error('You cannot reference an imported Subnet\'s route table if it was not supplied. Add the routeTableId when importing using Subnet.fromSubnetAttributes()')
+    }
```

Comment:
This would be a breaking change. With the old code, CDK users can do:
```ts
if (subnet.routeTable.routeTableId === undefined) {
  // handle gracefully
}
```

In addition, the old warning would warn once where this new change will warn every time `routeTable` is accessed.

-----

I went through the original issue and understood that the annoyance was because CDK users did not have a way to acknowledge the warning. But as mentioned in the issue, a feature was added to allow users doing so:
```
cdk.Annotations.of(subnet).acknowledgeWarning('@aws-cdk/aws-ec2:noSubnetRouteTableId');
```

With this feature, I think keeping the old behaviour is a less intrusive approach.